enhance: convert Message Touchable to React Native Gesture Handler

jjinendra3 opened the pull request:
<!-- This is a pull request template, you do not need to uncomment or remove the comments, they won't show up in the PR text. -->

## Proposed changes
<!-- Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue below. -->
Converted the message touchable to use React Native Gesture Handler (RNGH) instead of the default touchable.
Added left swipe to open thread functionality.
- Handled read-only rooms appropriately.
- Retained original onPress and onLongPress functionality while adding PanGestureHandler for smooth gestures.

Benefits of switching to RNGH:

- Improved performance – RNGH processes gestures in the native thread, making interactions more fluid.
- Better gesture handling – Prevents gesture conflicts and ensures smoother interactions across screens.
- More control over touch events – Supports advanced gestures like swiping while maintaining existing press behaviors.
## Issue(s)	
<!-- Link the issues being closed by or related to this PR. For example, you can use #594 if this PR closes issue number 594 -->
#6193 
## How to test or reproduce
<!-- Mention how you would reproduce the bug if not mentioned on the issue page already. Also mention which screens are going to have the changes if applicable -->
1. Open any chat room.
2. Try a left swipe on a message – it should open the thread.
3. Test long press and tap to ensure they still work as expected.
4. Try to left swipe a message inside thread room and it isn't enabled as expected.
5. Check in read-only rooms to confirm the swipe action isn't enabled as expected.

## Screenshots

https://github.com/user-attachments/assets/4d5a39c3-13fb-489d-ab4f-44410de093d1

## Types of changes
<!-- What types of changes does your code introduce to Rocket.Chat? -->
<!-- Put an `x` in the boxes that apply -->

- [x] Bugfix (non-breaking change which fixes an issue)
- [x] Improvement (non-breaking change which improves a current function)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation update (if none of the other choices apply)

## Checklist
<!-- Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code. -->

- [x] I have read the [CONTRIBUTING](https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat) doc
- [x] I have signed the [CLA](https://cla-assistant.io/RocketChat/Rocket.Chat.ReactNative)
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [ ] I have added necessary documentation (if applicable)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments
<!-- If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc... -->

jjinendra3 committed:
chore: converted Message Button to RNGH

jjinendra3 committed:
chore: added support for read only rooms

jjinendra3 committed:
chore: removed key

Rohit3523 commented on app/containers/message/Message.tsx:
```diff
@@ -172,7 +172,6 @@ const MessageTouchable = React.memo((props: IMessageTouchable & IMessage) => {
 	}
 	return (
 		<Touchable
-			key={props.id}
```
Removing key from existing code will break some test case

diegolmello commented on app/containers/message/MessageGestureHandler/Actions.tsx:
`containers/message/Components/${YOURS}`

diegolmello commented on app/containers/message/MessageGestureHandler/Touchable.tsx:
```diff
@@ -0,0 +1,139 @@
+	runOnJS
+} from 'react-native-reanimated';
+import {
+	LongPressGestureHandler,
+	PanGestureHandler,
```
GestureDetector

diegolmello reviewed (changes requested):
That's promising!

- Animation should look more like this https://telegram.tips/blog/swipe-left-to-reply/
- Missing haptic feedback when the threshold is reached

jjinendra3 committed:
chore: changed styling

jjinendra3 commented:
Added Haptics to press as well as PanGesture. Switched to GestureDetector and redesigned the thread swipe gesture.

If there are any discrepancies or if any refactor is needed, please do comment it.

https://github.com/user-attachments/assets/19bcd0b7-a8da-4988-afa1-2fbf670c6156

jjinendra3 committed:
Merge branch 'enhance/switchToRNGH' of https://github.com/jjinendra3/Rocket.Chat.ReactNative into enhance/switchToRNGH

jjinendra3 committed:
Merge branch 'develop' of https://github.com/jjinendra3/Rocket.Chat.ReactNative into enhance/switchToRNGH

jjinendra3 committed:
chore: needs fixing reactions

jjinendra3 committed:
chore: fixed issues
